hudi-client/hudi-client-common/src/main/java/org/apache/hudi/async/AsyncCompactService.java

```diff
@@ -81,8 +82,13 @@ protected Pair<CompletableFuture, ExecutorService> startService() {
 
           if (null != instant) {
             LOG.info("Starting Compaction for instant " + instant);
+            synchronized (writeConfigUpdateLock) {
+              compactor = createCompactor();
```

Comment:
We should only reinitialize the async compactor/clusterer when the write config is updated, not for every execution.

Reply:
+1